smartsim/_core/launcher/dragon/dragonBackend.py

```diff
@@ -95,31 +121,48 @@ def __init__(self, pid: int) -> None:
         self._queued_steps: "collections.OrderedDict[str, DragonRunRequest]" = (
             collections.OrderedDict()
         )
+        self._stop_requests: t.Deque[DragonStopRequest] = collections.deque()
         self._running_steps: t.List[str] = []
         self._completed_steps: t.List[str] = []
-
+        self._last_beat: float = 0.0
+        self._heartbeat()
+        self._last_update_time = self._last_beat
         num_hosts = len(self._hosts)
         host_string = str(num_hosts) + (" hosts" if num_hosts > 1 else " host")
         self._shutdown_requested = False
-        self._updates = 0
-        print(f"{host_string} available for execution: {self._hosts}")
+        self._can_shutdown = False
+        self._frontend_shutdown: t.Optional[bool] = None
+        logger.debug(f"{host_string} available for execution: {self._hosts}")
 
     def print_status(self) -> None:
-        print("\n-----------------------Launcher Status-----------------------")
-        print(f"| {self._updates}: System hosts: ", self._hosts)
-        print(f"| {self._updates}: Free hosts: ", list(self._free_hosts))
-        print(f"| {self._updates}: Allocated hosts: ", self._allocated_hosts)
-        print(f"| {self._updates}: Running steps: ", self._running_steps)
-        print(f"| {self._updates}: Group infos: ", self._group_infos)
-        print(f"| {self._updates}: There are {len(self._queued_steps)} queued steps")
-        print("-------------------------------------------------------------\n")
+        logger.debug(f"System hosts: {self._hosts}")
```

Comment:
seems like it would be misleading to come and try `mybackend.print_status()` only to discover you have to set your log level when it doesn't work.

If we rename to `get_status_message(self)` callers know what's happening.

`logger.info(get_status_message())` would work as expected, especially if you're in a REPL just trying to see what's happening, or if you decide to use that as the implementation of `__str__`

Reply:
Done done done